Framework/PythonInterface/test/python/mantid/utils/dgsTest.py

```diff
@@ -0,0 +1,462 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from numpy import array, nan, where, vstack
+from numpy.testing import assert_almost_equal, assert_array_equal
+from mantid.utils.dgs import qangle, ErrorCodes
+from mantid.geometry import OrientedLattice
+
+class DgsUtilsTest(unittest.TestCase):
+
+    def run_test(self, ol, expected_Qx, expected_Qy, expected_Qz, expected_in_plane_Q_angle, expected_out_plane_Q_angle,
```

Comment:
Consider some data structures to combine Qx, Qy and Qz, or combine expected_in_plane_Q_angel and expected_in_plane_kf_angle?

Reply:
I did. But it does not add anything useful for the user, and it will just make it more confusing. `result.Q_lab_x` is as good as `result.Q_lab.x`